src/citrine/gemd_queries/filter.py

```diff
@@ -0,0 +1,223 @@
+"""Definitions for GemdQuery objects, and their sub-objects."""
+from typing import List, Type
+
+from gemd.enumeration.base_enumeration import BaseEnumeration
+
+from citrine._serialization.serializable import Serializable
+from citrine._serialization.polymorphic_serializable import PolymorphicSerializable
+from citrine._serialization import properties
+
+__all__ = ['PropertyFilterType',
+           'RealFilter', 'IntegerFilter',
+           'MaterialClassification', 'TextSearchType',
+           ]
+
+
+class RealFilter(Serializable['RealFilter']):
```

Comment:
Like I mentioned previously, I don't think this logic is working at present, and I'd be hesitant to expose it now.
I think the only filters we have that we know work today are the All* filters.

Reply:
I've verified that at least the NominalRealFilter + RealFilter combo works, though the keys are all in camel case instead of snake case and if unit is the empty string, it erroneously gets dropped.  So buggy, but the results get filtered.

I'm comfortable not exposing it here if you have a high degree of confidence that there are no artifacts in the wild that use it.  Given the SDK's role in expert debugging, I don't want a table config fetch to result in a ValueError brick wall.

Reply:
"no artifacts in the wild that use it" - well, there's no SDK users that use it, I know that!

My understanding is that the UI folks tried it, found issues, requested a simpler approach (the All* filters) and use that exclusively now.

I've asked Lynnae to craft a bug report to get them fixed (or remove them), exposing this currently I think would be a bad thing.

Reply:
The risk of encountering objects being low and probably associated with Citrine team members, I'm going to not expose the filter classes that we are not planning on using moving forward.